status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import "sync"
+
+// Status represents a status update from an action within Empire.
+type Status interface {
+	// Returns a human readable string about the status update.
+	String() string
+}
+
+// status implements the Status interface
+type status struct {
+	message string
+}
+
+func (s *status) String() string {
+	return s.message
+}
+
+// NewStatus returns the default implementation of the Status interface
+func NewStatus(m string) *status {
+	return &status{message: m}
+}
+
+// StatusStream is an interface for publishing status updates while executing
+// an Empire action.
+type StatusStream interface {
+	// Publish publishes an update to the status stream
+	Publish(Status)
+
+	// Done finalizes the status stream
+	Done(error)
+}
+
+type SubscribableStream interface {
```

Comment:
Just food for thought. Another possibility here would be to forgo channels entirely and make the interface:

```go
// StatusStream is an interface for publishing status updates while executing
// an Empire action.
type StatusStream interface {
        // Publish publishes an update to the status stream
        Publish(Status) error

        // Done finalizes the status stream
        Done(error)

        // Wait returns a channel that that receives once Done() is called.
        // Consumers should call the Err() method to determine if an error
        // ocurred.
        Wait() <-chan struct{}

        // Returns the error from calling Done().
        Err() error
}
```

Then we just pass in an implementation that writes to the logstream:

```go
type jsonmessageStatusStream struct {
	sync.Mutex
	done chan struct{}
	err  error
	w    io.Writer
}

func (s *jsonmessageStream) Publish(message scheduler.Status) {
	select {
	case <-done:
		panic("Publish called on finalized stream")
	default:
	}

	json.NewEncoder(s.w).Encode(jsonmessage.Message{Status: status.Message})
}

func (s *jsonmessageStream) Done(err error) {
	close(s.done)
	s.err = err
}

func (s *jsonmessageStream) Err() error {
	return s.err
}

func (s *jsonmessageStream) Wait() <-chan struct{} {
	return s.done
}
```

Main advantage would be that we don't need to worry about buffering the channel, calls to `Publish` will immediately write to the stream and it's easy to wrap with middleware (e.g. writing to the apps kinesis stream, etc).